addons/blocked_locations/tests/test_location.py

```diff
@@ -113,4 +105,19 @@ def test09_onchange_blocked_to_stay_true(self):
 
         # simulate the triggering of the onchange, while remaining True
         self.test_location_01.onchange_u_blocked()
-        self.assertEqual(self.test_location_01.u_blocked_reason,'stock damaged')
\ No newline at end of file
+        self.assertEqual(self.test_location_01.u_blocked_reason,'stock damaged')
+
+    def test10_constrains_blocked(self):
+        """ Check that a bad combination of blocked state and reason not possible
+        """
+        self.test_location_01.u_blocked = True
+        self.test_location_01.u_blocked_reason = 'stock damaged'
+        msg = self.test_location_01._prepare_blocked_msg()
+        self.assertEqual(msg[0],
+                         'Location %s is blocked (reason: stock damaged).' %
+                         self.test_location_01.name)
```

Comment:
Should not assert the content of the error message. Could potentially assert that the phrase "stock damaged" is present in the error message, since the test case itself injects the "stock damaged phrase".